src/main/java/com/woodencloset/signalbot/SignalBot.java

```diff
@@ -78,6 +78,21 @@ public String getKeyStorePassword() {
     private List<Responder> responders = new LinkedList<>();
     private SignalServiceAccountManager accountManager;
 
+    public void registerWCaptcha(String username, RegistrationType type, String captcha) throws IOException, BackingStoreException {
+        logger.info("Sending verification code to " + username + ".");
+        prefs.clear();
+        String password = Base64.encodeBytes(Util.getSecretBytes(18));
+        prefs.put("LOCAL_USERNAME", username);
+        prefs.put("LOCAL_PASSWORD", password);
+        accountManager = new SignalServiceAccountManager(config, null, username, password, USER_AGENT);
+
+        if (type == RegistrationType.PhoneCall) {
+            accountManager.requestVoiceVerificationCode(Locale.getDefault(), Optional.absent(), Optional.absent());
+        } else {
+            accountManager.requestSmsVerificationCode(false,  Optional.fromNullable(captcha), Optional.absent());
+        }
+    }
+    
```

Comment:
Hi, can you please remove this function and instead add `String captcha` as an argument to the existing `register` method? You can then pass `null` as an argument when calling it in `Main.java` if the captcha argument wasn't specified. And the This will avoid code duplication. Also, I believe it could work for voice verification as well (although I'm not sure). Also - I think the correct syntax is Optional.ofNullable()